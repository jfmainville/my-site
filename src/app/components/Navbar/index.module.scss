@import "../../variables.module.scss";

.Main {
  display: flex;
  flex-direction: row-reverse;
  background-color: $background-color;
  padding: 2rem;
}

.Link {
  padding: 0 1rem 0 1rem;
}
