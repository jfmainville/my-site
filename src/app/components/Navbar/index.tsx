import styles from "./index.module.scss";

const Navbar = () => {
  return (
    <div className={styles.Navbar}>
      <div>Admin</div>
      <div>
        <button type="submit">
          <a href="/api/auth/logout">Logout</a>
        </button>
      </div>
    </div>
  );
};

export default Navbar;
