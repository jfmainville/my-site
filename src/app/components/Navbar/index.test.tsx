import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Navbar from "./index";

describe("<Navbar/> Test Suite", () => {
  it("should render the <Navbar/> component unchanged", () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });

  it("should contain the About link in the <Navbar/> component", async () => {
    render(<Navbar />);
    const button = screen.getByText("About");
    expect(button).toBeInTheDocument();
  });

  it("should contain the Experience link in the <Navbar/> component", async () => {
    render(<Navbar />);
    const button = screen.getByText("Experience");
    expect(button).toBeInTheDocument();
  });

  it("should contain the Blog link in the <Navbar/> component", async () => {
    render(<Navbar />);
    const button = screen.getByText("Blog");
    expect(button).toBeInTheDocument();
  });

  it("should contain the Contact link in the <Navbar/> component", async () => {
    render(<Navbar />);
    const button = screen.getByText("Contact");
    expect(button).toBeInTheDocument();
  });

  it("should contain the correct About link in the <Navbar/> component", async () => {
    render(<Navbar />);
    const button = screen.getByText("About");
    expect(button).toHaveAttribute("href", "/#about");
  });

  it("should contain the correct Experience link in the <Navbar/> component", async () => {
    render(<Navbar />);
    const button = screen.getByText("Experience");
    expect(button).toHaveAttribute("href", "/#experience");
  });

  it("should contain the correct Blog link in the <Navbar/> component", async () => {
    render(<Navbar />);
    const button = screen.getByText("Blog");
    expect(button).toHaveAttribute("href", "/#blog");
  });

  it("should contain the correct Contact link in the <Navbar/> component", async () => {
    render(<Navbar />);
    const button = screen.getByText("Contact");
    expect(button).toHaveAttribute("href", "/#contact");
  });
});
