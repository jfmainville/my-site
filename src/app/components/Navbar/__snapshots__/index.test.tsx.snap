// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> Test Suite should render the <Navbar/> component unchanged 1`] = `
<div>
  <div
    class="Navbar"
  >
    <div>
      Admin
    </div>
    <div>
      <button
        type="submit"
      >
        <a
          href="/api/auth/logout"
        >
          Logout
        </a>
      </button>
    </div>
  </div>
</div>
`;
